### ISSUE TITLE
Chapter 4: is a Property exactly equal to the set of ordered pairs?

### ISSUE THREAD

**Issue by Robert-Muil** [KGConf/Bookclub-DemystifyingOWL#27]
Is it strictly true to say that a Property *is* a set of ordered pairs? If we can assert properties of a Property, then it seems it is more than just the set of pairs?

I feel like this is analogous to Classes - you (Michael) were very careful not to equate Class and the set that it is associated with: is that for the same reason as here?

**Comment by Robert-Muil** [comment]
_Continuing conv from Zoom:_

Dean:
> Just like everything else in RDF, the property is identified by a URI.  When we assert properties of properties, we are using that URI in an ordered pair, which is an element of another set of ordered pairs, i.e., property

> A set absolutely can be Transitive; especially when that set is a set of ordered pairs.
A set of ordered pairs is transitive exactly if whenever (a,b) is in the set and (b,c) is in the set, then (a,c) is also in the set.

**Comment by Robert-Muil** [comment]
Dean, thank you! That suddenly clicks, I see what you mean: so in this case 'transitivity' is a description of the contents of the set, not some ephemeral characteristic _of_ it.

I was confused because mathematically a set is defined strictly by its members. Transitivity doesn’t have a meaning for what a set is, separate from its members.

I therefore erroneously concluded that it must simply be something imposed _upon_ a set: something about how that set is interpreted and used, but is not a _part_ of the set in any way.

However, your point is that it is neither some characteristic of the set that can be altered independently of its members, nor is it merely imposed upon it by use or interpretation: instead it is a _description of_ the contents of the set. The members of the set strictly define that set, and if the members fulfill transitivity, then we say the set is transitive.

**Comment by dallemang** [comment]
Exercise for the reader (if any reader wants to do this) - write out Functional, InverseFunctional, and Symmetric as descriptions of sets of ordered pairs like I did above for Transitive.  

Extra credit:  Look this up in the OWL spec and sort out the notation they use for this, and compare it to your answers (and mine, for that matter; I might have left out some special case).

**Comment by Robert-Muil** [comment]
My stab at it:

characteristic | set definition | informal definition
-------- | --------- | -------
Transitive | ∀a,b,c: _if_ (a,b)∈**S** _and_ (b,c)∈**S** _then_ (a,c)∈**S** | if (a,b) and (b,c) are in the set, then so is (a,c)
Functional |               ∀a,b,c: _if_ (a,b)∈**S** _and_ b≠c _then_ (a,c)∉**S** | if (a,b) is in the set then no other pair starting with a is: i.e. (a,c) is not in the set unless b=c
Inverse Functional | ∀a,b,c: _if_ (a,b)∈**S** _and_ a≠c _then_ (c,b)∉**S** | if (a,b) is in the set then no other pair ending with b is: i.e. (c,b) is not in the set unless a=c
Symmetric |             ∀a,b: _if_ (a,b)∈**S** _then_ (b,a)∈**S** | if (a,b) is in the set then so is (b,a)

**Comment by uscholdm** [comment]
@Robert-Muil 
> Is it strictly true to say that a Property is a set of ordered pairs? If we can assert properties of a Property, then it seems it is more than just the set of pairs?
> 
> I feel like this is analogous to Classes - you (Michael) were very careful not to equate Class and the set that it is associated with: is that for the same reason as here?

Yes, there is a close analogy with classes. A class _corresponds_ to a set of individuals. The correspondence is what defines the formal semantics of  a class in OWL. Similarly, a property _corresponds_ to a set of ordered pairs. That correspondence is what defines the formal semantics of a property in OWL. It is convenient, but not strictly true to think of properties and classes as _being_ the sets rather than merely _corresponding_ to them.  For every day modeling purposes, that difference is not important.

As far as whether something can be a set and also have a characteristic, think of a set of balls that have various colors.  There are a various statements that might or might not be true about any given set of colored balls.
1. all the balls are the same color
2. there are exactly two different colors of balls
3. no two balls are the same color
4. at least 3 balls are the same color
5. there is an odd number of balls
6. there is an odd number of different colors of balls 

If we have a particular set of 5 balls where 3 were red and 2 were blue, then which of the above statements 1-6 is true? 

This is just like asking what statements are true about a given set of orders pairs that correspond to a property. Possibilities include (from Dean's table):
1. if (a,b) and (b,c) are in the set, then so is (a,c)
2. if (a,b) is in the set then no other pair starting with a is: i.e. (a,c) is not in the set unless b=c
3. if (a,b) is in the set then no other pair ending with b is: i.e. (c,b) is not in the set unless a=c
4. if (a,b) is in the set then so is (b,a)

Asking whether a property is transitive, functional, inverse functional or symmetric is just like asking which of the 6 statements are true about the colored balls.  The difference is that statement 1-4 have mathematical and inference significance -- so much significance that mathematicians gave them names.  

**In summary, assigning a characteristic to a property is making an observation and a statement about what is true regarding the set of ordered pairs that corresponds to the property.**

**Comment by Robert-Muil** [comment]
Thank you Michael! BTW the table above is mine, so please don't assume it's correct!
